pytorch_lightning/callbacks/progress/progress.py

```diff
@@ -60,203 +52,29 @@ def format_num(n) -> str:
-        The :class:`~pytorch_lightning.trainer.trainer.Trainer` will call this to disable the
-        output on processes that have a rank different from 0, e.g., in multi-node training.
-        """
-        raise NotImplementedError
-
-    def enable(self):
-        """
-        You should provide a way to enable the progress bar.
-        The :class:`~pytorch_lightning.trainer.trainer.Trainer` will call this in e.g. pre-training
-        routines like the :ref:`learning rate finder <advanced/lr_finder:Learning Rate Finder>`
-        to temporarily enable and disable the main progress bar.
-        """
-        raise NotImplementedError
-
-    def print(self, *args, **kwargs):
-        """
-        You should provide a way to print without breaking the progress bar.
-        """
-        print(*args, **kwargs)
-
-    def on_init_end(self, trainer):
-        self._trainer = trainer
-
-    def on_train_start(self, trainer, pl_module):
-        self._train_batch_idx = trainer.fit_loop.batch_idx
-
-    def on_train_epoch_start(self, trainer, pl_module):
-        self._train_batch_idx = 0
-
-    def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        self._train_batch_idx += 1
-
-    def on_validation_start(self, trainer, pl_module):
-        self._val_batch_idx = 0
-
-    def on_validation_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        self._val_batch_idx += 1
-
-    def on_test_start(self, trainer, pl_module):
-        self._test_batch_idx = 0
-
-    def on_test_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        self._test_batch_idx += 1
-
-    def on_predict_epoch_start(self, trainer, pl_module):
-        self._predict_batch_idx = 0
-
-    def on_predict_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        self._predict_batch_idx += 1
-
-
 class ProgressBar(ProgressBarBase):
     r"""
     This is the default progress bar used by Lightning. It prints to `stdout` using the
     :mod:`tqdm` package and shows up to four different bars:
-
     - **sanity check progress:** the progress during the sanity check run
     - **main progress:** shows training + validation progress combined. It also accounts for
       multiple validation runs during training when
       :paramref:`~pytorch_lightning.trainer.trainer.Trainer.val_check_interval` is used.
     - **validation progress:** only visible during validation;
       shows total progress over all validation datasets.
     - **test progress:** only active when testing; shows total progress over all test datasets.
-
     For infinite datasets, the progress bar never ends.
-
     If you want to customize the default ``tqdm`` progress bars used by Lightning, you can override
     specific methods of the callback class and pass your custom implementation to the
     :class:`~pytorch_lightning.trainer.trainer.Trainer`:
-
```

Comment:
where is this coming from? your auto-formatter definitely does not do the right thing, these spaces are necessary. @tchaton has the same problem. consider turning it off.

Reply:
Resolved! Thanks for the catch ^^